### DISCUSSION TITLE
Deploying via scripts and to NOT USE .env

### DISCUSSION THREAD

**Discussion by toastdoteth2** [Cyfrin/foundry-full-course-cu#2266] [General]
Just now watching Foundry Fundamentals about private key security.

Wouldnt the better advice to just use a hardware wallet and not use private keys at all? Understand that using scripts makes testing easier but when it comes to a production deployment then perhaps revert to `forge create` which allows a hardware wallet signature?

**Comment by EngrPips** [comment]
Honestly, Considering how much careful and sensitive we want to be with details such as private keys during deployment then, using a Hardware wallet seems like a good option if a keystore wallet does not suffice for us.
